File: java/src/sdk/dataobjects/paypal/payflow/MagTekInfo.java
package paypal.payflow;

/**
 * Used for Magtek related information.
 * <p>
 * Use the MagtekInfo object for the Magtek
 * encrypted card reader related information.
 * <p>
 * Following example shows how to use the MagtekInfo object.
 * <p>
 * // Swipe is the SwipeCard object
 */
public final class MagTekInfo extends BaseRequestDataObject {

    private String encMP;
    private String encryptionBlockType;
    private String encTrack1;
    private String encTrack2;
    private String encTrack3;
    private String ksn;
    private String magtekCardType;
    private String mpStatus;
    private String registeredBy;
    private String swipedECRHost;
    private String deviceSN;
    private String merchantId;
    private String pan4;
    private String pCode;
    private String authValue1;
    private String authValue2;
    private String authValue3;
    private String magtekUserName;
    private String magtekPassword;

    /**
     * Constructor
     * <p>Use this class to set the MagTek related information. Used for encrypted swipe.</p>
     */

    public MagTekInfo() {
    }

    protected void generateRequest() {
        try {
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_ENCMP, encMP));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_ENCRYPTIONBLOCKTYPE, encryptionBlockType));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_ENCTRACK1, encTrack1));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_ENCTRACK2, encTrack2));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_ENCTRACK3, encTrack3));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_KSN, ksn));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_MAGTEKCARDTYPE, magtekCardType));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_REGISTEREDBY, registeredBy));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_SWIPEDECRHOST, swipedECRHost));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_DEVICESN, deviceSN));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_MPSTATUS, mpStatus));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_PAN4, pan4));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_MERCHANTID, merchantId));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_PCODE, pCode));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_AUTHVALUE1, authValue1));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_AUTHVALUE2, authValue2));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_AUTHVALUE3, authValue3));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_MAGTEKUSERNAME, magtekUserName));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.MAGTEK_PARAM_MAGTEKPWD, magtekPassword));


        } catch (Exception ex) {
            ErrorObject err = new ErrorObject(PayflowConstants.SEVERITY_FATAL, "", ex.toString());
            if (getContext() != null) {
                getContext().addError(err);
            }
        }
    }

    /**
     * Gets the encrypted MagnePrint information.
     *
     * @return encMP String
     * <p>Maps to Payflow Parameter: ENCMP</p>
     */
    public String getEncMP() {
        return encMP;
    }

    /**
     * Sets the encrypted MagnePrint information.
     *
     * @param encMP String
     * <p>Maps to Payflow Parameter: ENCMP</p>
     */
    public void setEncMP(String encMP) {
        this.encMP = encMP;
    }

    /**
     * Gets the Encryption Block Type
     *
     * @return encryptionBlockType String
     * <p>Maps to Payflow Parameter: ENCRYPTIONBLOCKTYPE</p>
     */
    public String getEncryptionBlockTyp() {
        return encryptionBlockType;
    }

    /**
     * Sets the Encryption Block Type
     *
     * @param encryptionBlockType String
     * <p>Maps to Payflow Parameter: ENCRYPTIONBLOCKTYPE</p>
     */
    public void setEncryptionBlockType(String encryptionBlockType) {
        this.encryptionBlockType = encryptionBlockType;
    }

    /**
     * Gets the Encrypted Track 1
     *
     * @return encTrack1 String
     * <p>Maps to Payflow Parameter: ENCTRACK1</p>
     */
    public String getEncTrack1() {
        return encTrack1;
    }

    /**
     * Sets the Encrypted Track 1
     *
     * @param encTrack1 String
     * <p>Maps to Payflow Parameter: ENCTRACK1</p>
     */
    public void setEncTrack1(String encTrack1) {
        this.encTrack1 = encTrack1;
    }

    /**
     * Gets the Encrypted Track 2
     *
     * @return encTrack2 String
     * <p>Maps to Payflow Parameter: ENCTRACK2</p>
     */
    public String getEncTrack2() {
        return encTrack2;
    }

    /**
     * Sets the Encrypted Track 2
     *
     * @param encTrack2 String
     * <p>Maps to Payflow Parameter: ENCTRACK2</p>
     */
    public void setEncTrack2(String encTrack2) {
        this.encTrack2 = encTrack2;
    }

    /**
     * Gets the Encrypted Track 3
     *
     * @return encTrack3 String
     * <p>Maps to Payflow Parameter: ENCTRACK3</p>
     */
    public String getEncTrack3() {
        return encTrack3;
    }

    /**
     * Sets the Encrypted Track 3
     *
     * @param encTrack3 String
     * <p>Maps to Payflow Parameter: ENCTRACK3</p>
     */
    public void setEncTrack3(String encTrack3) {
        this.encTrack3 = encTrack3;
    }

    /**
     * Gets the KSN.
     *
     * @return ksn String
     * <p>Maps to Payflow Parameter: KSN</p>
     */
    public String getKsn() {
        return ksn;
    }

    /**
     * Sets the KSN.
     *
     * @param ksn String
     * <p>Maps to Payflow Parameter: KSN</p>
     */
    public void setKsn(String ksn) {
        this.ksn = ksn;
    }

    /**
     * Gets the MagTek Card Type
     *
     * @return magtekCardType String
     * <p>Maps to Payflow Parameter: MAGTEKCARDTYPE</p>
     */
    public String getMagtekCardType() {
        return magtekCardType;
    }

    /**
     * Sets the MagTek Card Type.
     *
     * @param magtekCardType String
     * <p>Maps to Payflow Parameter: MAGTEKCARDTYPE</p>
     */
    public void setMagtekCardType(String magtekCardType) {
        this.magtekCardType = magtekCardType;
    }

    /**
     * Gets the MP Status.
     *
     * @return mpStatus String
     * <p>Maps to Payflow Parameter: MPSTATUS</p>
     */
    public String getMpStatus() {
        return mpStatus;
    }

    /**
     * Sets the MP Status.
     *
     * @param mpStatus String
     * <p>Maps to Payflow Parameter: MPSTATUS</p>
     */
    public void setMpStatus(String mpStatus) {
        this.mpStatus = mpStatus;
    }

    /**
     * Gets the Registerd By.
     *
     * @return registeredBy String
     * <p>Maps to Payflow Parameter: REGISTEREDBY</p>
     */
    public String getRegisteredBy() {
        return registeredBy;
    }

    /**
     * Sets the Registered By.
     *
     * @param registeredBy String
     * <p>Maps to Payflow Parameter: REGISTEREDBY</p>
     */
    public void setRegisteredBy(String registeredBy) {
        this.registeredBy = registeredBy;
    }

    /**
     * Gets the Swiped ECR Host.
     *
     * @return swipedECRHost String
     * <p>Maps to Payflow Parameter: SWIPEDECRHOST</p>
     */
    public String getSwipedECRHost() {
        return swipedECRHost;
    }

    /**
     * Sets the Swiped ECR Host.
     *
     * @param swipedECRHost String
     * <p>Maps to Payflow Parameter: SWIPEDECRHOST</p>
     */
    public void setSwipedECRHost(String swipedECRHost) {
        this.swipedECRHost = swipedECRHost;
    }

    /**
     * Gets the Device Serial Number.
     *
     * @return deviceSN String
     * <p>Maps to Payflow Parameter: DEVICESN</p>
     */
    public String getDeviceSN() {
        return deviceSN;
    }

    /**
     * Sets the Device Serial Number.
     *
     * @param deviceSN String
     * <p>Maps to Payflow Parameter: DEVICESN</p>
     */
    public void setDeviceSN(String deviceSN) {
        this.deviceSN = deviceSN;
    }

    /**
     * Gets the Merchant's ID.
     *
     * @return merchantId String
     * <p>Maps to Payflow Parameter: MERCHANTID</p>
     */
    public String getMerchantId() {
        return merchantId;
    }

    /**
     * Sets the Merchant's ID.
     *
     * @param merchantId String
     * <p>Maps to Payflow Parameter: MERCHANTID</p>
     */
    public void setMerchantId(String merchantId) {
        this.merchantId = merchantId;
    }

    /**
     * Gets the last 4-digits of the PAN.
     *
     * @return pan4 String
     * <p>Maps to Payflow Parameter: PAN4</p>
     */
    public String getPan4() {
        return pan4;
    }

    /**
     * Sets the last 4-digits of the PAN.
     *
     * @param pan4 String
     * <p>Maps to Payflow Parameter: PAN4</p>
     */
    public void setPan4(String pan4) {
        this.pan4 = pan4;
    }

    /**
     * Gets the generated Protection Code.
     *
     * @return pCode String
     * <p>Maps to Payflow Parameter: PCODE</p>
     */
    public String getpCode() {
        return pCode;
    }

    /**
     * Sets the generated Protection Code.
     *
     * @param pCode String
     * <p>Maps to Payflow Parameter: PCODE</p>
     */
    public void setpCode(String pCode) {
        this.pCode = pCode;
    }

    /**
     * Gets the Authentication Value 1 generated with the PCode.
     *
     * @return authValue1; String
     * <p>Maps to Payflow Parameter: AUTHVALUE1</p>
     */
    public String getAuthValue1() {
        return authValue1;
    }

    /**
     * Sets the Authentication Value 1 generated with the PCode.
     *
     * @param authValue1; String
     * <p>Maps to Payflow Parameter: AUTHVALUE1</p>
     */
    public void setAuthValue1(String authValue1) {
        this.authValue1 = authValue1;
    }

    /**
     * Gets the Authentication Value 2 generated with the PCode.
     *
     * @return authValue2 String
     * <p>Maps to Payflow Parameter: AUTHVALUE2</p>
     */
    public String getAuthValue2() {
        return authValue2;
    }

    /**
     * Sets the Authentication Value 2 generated with the PCode.
     *
     * @param authValue2 String
     * <p>Maps to Payflow Parameter: AUTHVALUE2</p>
     */
    public void setAuthValue2(String authValue2) {
        this.authValue2 = authValue2;
    }

    /**
     * Gets the Authentication Value 3 generated with the PCode.
     *
     * @return authValue3 String
     * <p>Maps to Payflow Parameter: AUTHVALUE3</p>
     */
    public String getAuthValue3() {
        return authValue3;
    }

    /**
     * Sets the Authentication Value 3 generated with the PCode.
     *
     * @param authValue3 String
     * <p>Maps to Payflow Parameter: AUTHVALUE3</p>
     */
    public void setAuthValue3(String authValue3) {
        this.authValue3 = authValue3;
    }

    /**
     * Gets the MagTek User Name.
     *
     * @return magtekUserName String
     * <p>Maps to Payflow Parameter: MAGTEKUSERNAME</p>
     */
    public String getMagtekUserName() {
        return magtekUserName;
    }

    /**
     * Sets the MagTek User Name.
     *
     * @param magtekUserName String
     * <p>Maps to Payflow Parameter: MAGTEKUSERNAME</p>
     */
    public void setMagtekUserName(String magtekUserName) {
        this.magtekUserName = magtekUserName;
    }

    /**
     * Gets the MagTek Password.
     *
     * @return merchSvc String
     * <p>Maps to Payflow Parameter: MAGTEKPASSWORD</p>
     */
    public String getMagtekPassword() {
        return magtekPassword;
    }

    /**
     * Sets the MagTek Password.
     *
     * @param magtekPassword String
     * <p>Maps to Payflow Parameter: MAGTEKPASSWORD</p>
     */
    public void setMagtekPassword(String magtekPassword) {
        this.magtekPassword = magtekPassword;
    }
}

File: dotNET/PayflowProSDK/PayPal/Payments/DataObjects/MagtekInfo.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{
    /// <summary>
    /// Used for Magtek related information.
    /// </summary>
    /// <remarks>Use the MagtekInfo object for the Magtek
    /// encrypted card reader related information.</remarks>
    /// <example>
    /// <para>Following example shows how to use the MagtekInfo object.</para>
    /// <code lang="C#" escaped="false">
    ///	.................
    ///	// Swipe is the SwipeCard object
    ///	.................
    ///	// Set the Magtek Info details.
    /// MagtekInfo MT = new MagtekInfo();
    /// MT.DeviceSN = "Device Serial Number from reader";
    /// MT.EncMP = "ENCMP from reader";
    /// MT.EncryptionBlockType = "1";
    /// MT.EncTrack1 = "Track 1 data from reader";
    /// MT.EncTrack2 = "Track 2 data from reader";
    /// MT.EncTrack3 = "";
    /// MT.KSN = "KSN from reader";
    /// MT.MagtekCardType = "1";
    /// MT.MPStatus = "MPStatus from reader";
    /// MT.RegisteredBy = "PayPal";
    /// MT.SwipedECRHost = "MAGT";
    /// // When using Encrypted Card Readers you do not populate the SwipeCard object as the data from the Magtek object
    /// // will be used instead.
    /// SwipeCard Swipe = new SwipeCard("");
    /// Swipe.MagtekInfo = MT;
    ///	.................
    ///	</code>
    /// <code lang="Visual Basic" escaped="false">
    ///	.................
    ///	' Inv is the Invoice object
    ///	.................
    ///	' Set the Browser Info details.
    ///	Dim Browser As BrowserInfo = New BrowserInfo
    ///	Browser.BrowserCountryCode  = "USA"
    ///	Browser.BrowserUserAgent = "IE 6.0"
    ///	Inv.BrowserInfo = Browser
    ///	.................
    ///	</code>
    /// </example>
    public sealed class MagtekInfo : BaseRequestDataObject
    {
        #region "Member Variables"

        /// <summary>
        /// Holds Encrypted MagnePrint Information
        /// </summary>
        private String mEncMP;
        /// <summary>
        /// Holds Encryption Block
        /// </summary>
        private String mEncryptionBlockType;
        /// <summary>
        /// Holds Encrypted Track 1 
        /// </summary>
        private String mEncTrack1;
        /// <summary>
        /// Holds Encrypted Track 2
        /// </summary>
        private String mEncTrack2;
        /// <summary>
        /// Holds Encrypted Track 3
        /// </summary>
        private String mEncTrack3;
        /// <summary>
        /// Holds KSN
        /// </summary>
        private String mKSN;
        /// <summary>
        /// Holds Card Type
        /// /// </summary>
        private String mMagtekCardType;
        /// <summary>
        /// Holds MP Status
        /// </summary>
        private String mMPStatus;
        /// <summary>
        /// Holds Registered By
        /// </summary>
        private String mRegisteredBy;
        /// <summary>
        /// Holds Swiped ECR Host
        /// </summary>
        private String mSwipedECRHost;
        /// <summary>
        /// Holds Device Serial Number
        /// </summary>
        private String mDeviceSN;
        /// <summary>
        /// Holds Merchant Id
        /// </summary>
        private String mMerchantId;
        /// <summary>
        /// Holds 4-digit PAN
        /// </summary>
        private String mPAN4;
        /// <summary>
        /// Holds Protection Code
        /// </summary>
        private String mPCode;
        /// <summary>
        /// Holds Authorization Value 1
        /// </summary>
        private String mAuthValue1;
        /// <summary>
        /// Holds Authorization Value 2
        /// </summary>
        private String mAuthValue2;
        /// <summary>
        /// Holds Authorization Value 3
        /// </summary>
        private String mAuthValue3;
        /// <summary>
        /// Holds Magtek User Name
        /// </summary>
        private String mMagtekUserName;
        /// <summary>
        /// Holds Magtek Password
        /// </summary>
        private String mMagtekPassword;

        #endregion

        #region "Constructors"

        /// <summary>
        /// Constructor for BrowserInfo
        /// </summary>
        /// <remarks>Use the MagtekInfo object for the Magtek
        /// encrypted card reader related information.</remarks>
        /// <example>
        /// <para>Following example shows how to use a 
        /// Magtek Info object.</para>
        /// <code lang="C#" escaped="false">
        ///	.................
        ///	// Swipe is the SwipeCard object
        ///	.................
        /// // Set the MagtekInfo object.
        /// MagtekInfo MT = new MagtekInfo();
        /// MT.DeviceSN = "Device Serial Number from reader";
        /// MT.EncMP = "ENCMP from reader";
        /// MT.EncryptionBlockType = "1";
        /// MT.EncTrack1 = "Track 1 data from reader";
        /// // When using Encrypted Card Readers you do not populate the SwipeCard object as the data from the Magtek object
        /// // will be used instead.
        /// SwipeCard Swipe = new SwipeCard("");
        /// Swipe.MagtekInfo = MT;
        ///	.................
        ///	.................
        ///	</code>
        /// <code lang="Visual Basic" escaped="false">
        ///	.................
        ///	' Inv is the Invoice object
        ///	.................
        ///	' Set the Browser Info details.
        ///	Dim Browser As BrowserInfo = New BrowserInfo
        ///	Browser.BrowserCountryCode  = "USA"
        ///	Browser.BrowserUserAgent = "IE 6.0"
        ///	Inv.BrowserInfo = Browser
        ///	.................
        ///	</code>
        /// </example>
        public MagtekInfo()
        {
        }

        #endregion

        #region "Properties"

        /// <summary>
        /// Gets, Sets Magtek Encrypted MagnePrint Information.
        /// </summary>
        /// <remarks>
        /// <para>Encrypted MagnePrint Information returned by a MagneSafe device when a card is swiped.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>ENCMP</code>
        /// </remarks>
        public String EncMP
        {
            get { return mEncMP; }
            set { mEncMP = value; }
        }

        /// <summary>
        /// Gets, Sets Magtek Encryption Block.
        /// </summary>
        /// <remarks>
        /// <para>The code which indicates what type of Encryption Block is used.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>ENCRYPTIONBLOCKTYPE</code>
        /// </remarks>
        public String EncryptionBlockType
        {
            get { return mEncryptionBlockType; }
            set { mEncryptionBlockType = value; }
        }

        /// <summary>
        /// Gets, Sets Magtek Encrypted Track 1.
        /// </summary>
        /// <remarks>
        /// <para>Encrypted Track 1 information returned by a MagneSafe device when a card is swiped.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>ENCTRACKE1</code>
        /// </remarks>
        public String EncTrack1
        {
            get { return mEncTrack1; }
            set { mEncTrack1 = value; }
        }
        /// <summary>
        /// Gets, Sets Magtek Encrypted Track 2.
        /// </summary>
        /// <remarks>
        /// <para>Encrypted Track 2 information returned by a MagneSafe device when a card is swiped.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>ENCTRACKE2</code>
        /// </remarks>
        public String EncTrack2
        {
            get { return mEncTrack2; }
            set { mEncTrack2 = value; }
        }
        /// <summary>
        /// Gets, Sets Magtek Encrypted Track 3.
        /// </summary>
        /// <remarks>
        /// <para>Encrypted Track 3 information returned by a MagneSafe device when a card is swiped.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>ENCTRACKE2</code>
        /// </remarks>
        public String EncTrack3
        {
            get { return mEncTrack3; }
            set { mEncTrack3 = value; }
        }

        /// <summary>
        /// Gets or Sets Magtek KSN
        /// </summary>
        /// <remarks>
        /// <para>20 character string returned by a MagneSafe device when a card is swiped.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>KSN</code>
        /// </remarks>
        public String KSN
        {
            get { return mKSN; }
            set { mKSN = value; }
        }

        /// <summary>
        /// Gets or Sets Magtek Card Data Format
        /// </summary>
        /// <remarks>
        /// <para>The code which indicates what type of Card Data Format is being submitted.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MAGTEKCARDTYPE</code>
        /// </remarks>
        public String MagtekCardType
        {
            get { return mMagtekCardType; }
            set { mMagtekCardType = value; }
        }

        /// <summary>
        /// Gets or Sets Magtek MagnePrint Status
        /// </summary>
        /// <remarks>
        /// <para>MagnePrint Status of Card Swipe. This is an alpha numeric string, returned by a MagneSafe device when a card is swiped.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MPSTATUS</code>
        /// </remarks>
        public String MPStatus
        {
            get { return mMPStatus; }
            set { mMPStatus = value; }
        }
        /// <summary>
        /// Gets or Sets Magtek Registered By
        /// </summary>
        /// <remarks>
        /// <para>An alpha numeric entry between 1 and 20 characters long.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>REGISTEREDBY</code>
        /// </remarks>
        public String RegisteredBy
        {
            get { return mRegisteredBy; }
            set { mRegisteredBy = value; }
        }
        /// <summary>
        /// Gets or Sets Magtek Swipe ECR Host
        /// </summary>
        /// <remarks>
        /// <para>MAGT is the only value that is accepted in the SWIPEDECRHOST parameter.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>SWIPEDECRHOST</code>
        /// </remarks>
        public String SwipedECRHost
        {
            get { return mSwipedECRHost; }
            set { mSwipedECRHost = value; }
        }

        /// <summary>
        /// Gets or Sets Magtek device serial number
        /// </summary>
        /// <remarks>
        /// <para>The device serial number of the Magtek card reader.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>DEVICESN</code>
        /// </remarks>
        public String DeviceSN
        {
            get { return mDeviceSN; }
            set { mDeviceSN = value; }
        }
        /// <summary>
        /// Gets or Sets Magtek Merchant Id
        /// </summary>
        /// <remarks>
        /// <para>Your Merchant ID or the Merchant ID of the merchant redeeming the Protection Code.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTID</code>
        /// </remarks>
        public String MerchantId
        {
            get { return mMerchantId; }
            set { mMerchantId = value; }
        }
        /// <summary>
        /// Gets or Sets Magtek PAN4
        /// </summary>
        /// <remarks>
        /// <para>The last four digits of the PAN / account number encoded in the card.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>PAN4</code>
        /// </remarks>
        public String PAN4
        {
            get { return mPAN4; }
            set { mPAN4 = value; }
        }
        /// <summary>
        /// Gets or Sets Magtek Protection Code
        /// </summary>
        /// <remarks>
        /// <para>The generated Protection Code.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>PCODE</code>
        /// </remarks>
        public String PCode
        {
            get { return mPCode; }
            set { mPCode = value; }
        }
        /// <summary>
        /// Gets or Sets Magtek Authentication Value 1
        /// </summary>
        /// <remarks>
        /// <para>Authentication Value 1 generated with the PCode.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>AUTHVALUE1</code>
        /// </remarks>
        public String AuthValue1
        {
            get { return mAuthValue1; }
            set { mAuthValue1 = value; }
        }
        /// <summary>
        /// Gets or Sets Magtek Authentication Value 2
        /// </summary>
        /// <remarks>
        /// <para>Authentication Value 2 generated with the PCode.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>AUTHVALUE2</code>
        /// </remarks>
        public String AuthValue2
        {
            get { return mAuthValue2; }
            set { mAuthValue2 = value; }
        }
        /// <summary>
        /// Gets or Sets Magtek Authentication Value 3
        /// </summary>
        /// <remarks>
        /// <para>Authentication Value 3 generated with the PCode.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>AUTHVALUE3</code>
        /// </remarks>
        public String AuthValue3
        {
            get { return mAuthValue3; }
            set { mAuthValue3 = value; }
        }
        /// <summary>
        /// Gets or Sets MagTek User Name
        /// </summary>
        /// <remarks>
        /// <para> MagTek user name.</para> 
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MAGTEKUSERNAME</code>
        /// </remarks>
        public String MagtekUserName
        {
            get { return mMagtekUserName; }
            set { mMagtekUserName = value; }
        }
        /// <summary>
        /// Gets or Sets Magtek password
        /// </summary>
        /// <remarks>
        /// <para>Magtek password.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MAGTEKPWD</code>
        /// </remarks>
        public String MagtekPassword
        {
            get { return mMagtekPassword; }
            set { mMagtekPassword = value; }
        }
        #endregion

        #region "Core functions"

        /// <summary>
        /// Generates the transaction request.
        /// </summary>
        internal override void GenerateRequest()
        {
            try
            {
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_ENCMP, mEncMP));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_ENCRYPTIONBLOCKTYPE, mEncryptionBlockType));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_ENCTRACK1, mEncTrack1));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_ENCTRACK2, mEncTrack2));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_ENCTRACK3, mEncTrack3));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_KSN, mKSN));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_MAGTEKCARDTYPE, mMagtekCardType));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_REGISTEREDBY, mRegisteredBy));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_SWIPEDECRHOST, mSwipedECRHost));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_DEVICESN, mDeviceSN));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_MPSTATUS, mMPStatus));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_MERCHANTID, mMerchantId));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_PAN4, mPAN4));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_PCODE, mPCode));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_AUTHVALUE1, mAuthValue1));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_AUTHVALUE2, mAuthValue2));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_AUTHVALUE3, mAuthValue3));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_MAGTEKUSERNAME, mMagtekUserName));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.MAGTEK_PARAM_MAGTEKPWD, mMagtekPassword));
            }
            catch (BaseException)
            {
                throw;
            }
            catch (Exception Ex)
            {
                DataObjectException DEx = new DataObjectException(Ex);
                throw DEx;
            }
            //catch
            //{
            //    throw new Exception();				
            //}
        }
        #endregion
    }
}